### PR TITLE
MWPW-147683 - CSS Cleanup

### PR DESCRIPTION
**Pull request by colloyd**
* Final pass on the css
* Edits to markup where necessary

Resolves: [MWPW-147683](https://jira.corp.adobe.com/browse/MWPW-147683)

**Test URLs:**
- Before:  https://quiz-entry-block--milo--adobecom.hlx.page/drafts/colloyd/quiz-entry/?martech=off
- After: https://card-and-input-css--milo--colloyd.hlx.page/drafts/colloyd/quiz-entry/?martech=off

Testing notes:
The css should match the comp (find the link in the Jira ticket) in mobile, tablet, and portrait. This includes card states (selected, keyboard focus, etc.), the ML input, with it's autocomplete box and all of the text sizes and spacing. I did do some work to bring the carousel more in line. I'm not sure how animating it is possible, given how the cards are dynamically added or removed when the slides are advanced, so that not present. Finally, I tested this in FF and Safari to be sure that everything  holds up and it does.

### CODE REVIEW

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 1]
```diff
@@ -1,142 +1,243 @@
...
-/* Core layout for the quiz entry */
-.quiz-entry-container {
-  display: flex;
-  flex-direction: column;
+.quiz-container {
   align-items: center;
   color: var(--color-black);
+  display: flex;
+  flex-direction: column;
+  padding: 80px 48px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:6]: --spacing-xxxl, --spacing-xl

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 2]
```diff
@@ -1,142 +1,243 @@
...
-/* Styles for titles and subtitles with centralized text */
-.quiz-entry-title, .quiz-entry-subtitle {
-  text-align: center;
+.quiz-container *{
+  box-sizing: border-box;
+}
+
+.quiz-heading-container {
   color: #2C2C2C;
+  margin-bottom: 32px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:15]: --spacing-m

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 3]
```diff
@@ -1,142 +1,243 @@
...
-  padding-left: 20px;
-  width: 305px;
-  height: 56px;
-  cursor: pointer;
-  box-sizing: border-box;
+.quiz-question-container {
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+  margin-bottom: 56px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:35]: --spacing-xxl

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 4]
```diff
@@ -1,142 +1,243 @@
...
+  background-position: center right 16px;
+  border: 2px solid #e6e6e6;
+  border-radius: 40px;
+  color: #2c2c2c;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:54]: var(--text-color)

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 5]
```diff
@@ -1,142 +1,243 @@
...
+  cursor: pointer;
+  font-size: 16px;
+  height: 56px;
+  line-height: 24px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:58]: --type-heading-m-lh

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 6]
```diff
@@ -1,142 +1,243 @@
...
+  padding: 0 56px 0 20px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:59]: --spacing-xl, --spacing-s?

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 7]
```diff
@@ -1,142 +1,243 @@
...
+}
+
+.popover-content div:hover {
+  background-color: #f0f0f0; /* Highlight color on hover */
+  text-decoration: underline;
+}
+
+.popover-item {
+  cursor: pointer;
+  font-size: 16px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:128]: --type-detail-l-size?

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 8]
```diff
@@ -1,142 +1,243 @@
...
+  font-weight: 400;
+  line-height: 24px;
+  color: #1473E6;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:131]: --link-color-dark

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 9]
```diff
@@ -1,142 +1,243 @@
...
     color: #FFF;
+    display: flex;
     font-size: 16px;
     font-weight: 900;
+    height: 40px;
+    justify-content: center;
     line-height: 30px;
-    margin-bottom: 8px;
-    margin-top: 32px;
+    margin: 32px 0 8px;
```
Comment (fullcolorcoder) [inline libs/blocks/quiz-entry/quiz-entry.css:145]: --spacing-m

**Review by fullcolorcoder — CHANGES_REQUESTED** [review]
Vars, also a small eslint spacing thing

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 10]
```diff
@@ -248,26 +361,29 @@ html[dir="rtl"] .quiz-option-icon {
...
   display: flex;
   justify-content: center;
-  margin: 50px 0 24px;
 }
 
 .quiz-button {
   background: var(--color-gray-800);
-  border: 2px solid var(--color-white);
-  border-radius: 30px;
+  border: none;
```
Comment (JackySun9) [inline libs/blocks/quiz-entry/quiz-entry.css:367]: border: 2px

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 11]
```diff
@@ -248,26 +361,29 @@ html[dir="rtl"] .quiz-option-icon {
...
+}
+
+.quiz-button:focus-visible {
+  outline: 4px solid #4B75FF;
+  outline-offset: 2px;
 }
 
 .quiz-button-label {
   color: var(--color-white);
   font-size: 18px;
```
Comment (JackySun9) [inline libs/blocks/quiz-entry/quiz-entry.css:384]: font-size: 17px
line-height: 21px

**Review by JackySun9 — CHANGES_REQUESTED** [review]

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 12]
```diff
@@ -1,142 +1,243 @@
...
+  max-width: 1130px;
+  width: 100%;
 }
 
-.ml-field-container {
+.quiz-input-container {
+  display: flex;
+  flex-flow:row nowrap;
+  justify-content: center;
+  max-width: 480px;
```
[the remark this answers is not in the document]
Comment (colloyd) [reply libs/blocks/quiz-entry/quiz-entry.css:44]: I found that there are a lot places in the figma design that used odd numbers like 489px, etc. I've rounded all of these out to more sane numbers where it made sense, like this one.

**Comment by JackySun9** [comment]
Screenshot 2024-05-07 at 11 39 23 AM
When I select the card, I found the while small stuff on the top right corner of the card, do you know why?

File: libs/blocks/quiz-entry/quiz-entry.css (hunk 11 above)
Comment (colloyd) [reply libs/blocks/quiz-entry/quiz-entry.css:384]: again, this is a spot where standard milo / consonant values make sense.

**Comment by colloyd** [comment]
> Screenshot 2024-05-07 at 11 39 23 AM When I select the card, I found the while small stuff on the top right corner of the card, do you know why?

Because it's in the figma design for button selected states.

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 13]
```diff
@@ -1,142 +1,238 @@
...
+  padding: 8px;
+  position: relative;
+  width: 100%;
 }
 
-.carousel-slides {
-  display: flex;
+.carousel-arrow {
+  background-color: #fff;
+  background-image: url('https://main--milo--adobecom.hlx.page/libs/blocks/carousel/img/arrow.svg');
```
Comment (JackySun9) [inline libs/blocks/quiz-entry/quiz-entry.css:174]: the color of arrow should be black

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 14]
```diff
@@ -1,142 +1,238 @@
...
-  display: block;
-  position: absolute;
-  right: 13px;
-  top: calc(50% - 16px);
-  cursor: pointer;
-  width: 32px;
-  height: 32px;
+.quiz-input-clear {
+  background: url("data:image/svg+xml,%0A%3Csvg id='Layer_1' data-name='Layer 1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 20'%3E%3Crect id='Frame' x='4.14' y='3.77' width='12' height='12' style='fill:red;fill-opacity:0'/%3E%3Cg id='Line'%3E%3Cpath d='M14.88,13.16a1,1,0,0,1,0,1.35,1,1,0,0,1-.67.27.94.94,0,0,1-.67-.27l-3.4-3.4-3.39,3.4a1,1,0,0,1-1.35,0,1,1,0,0,1,0-1.35L8.8,9.77,5.4,6.37A1,1,0,0,1,5.4,5,1,1,0,0,1,6.75,5l3.39,3.39L13.54,5a.95.95,0,1,1,1.34,1.34l-3.39,3.4Z' style='fill:%236e6e6e'/%3E%3C/g%3E%3C/svg%3E") center no-repeat #fff;
```
Comment (JackySun9) [inline libs/blocks/quiz-entry/quiz-entry.css:81]: please add same position like search icon, same distance to the right side

**Review by JackySun9 — CHANGES_REQUESTED** [review]

**Comment by JackySun9** [comment]
Screenshot 2024-05-07 at 1 38 49 PM
I see two border here, can we have only one?

**Comment by colloyd** [comment]
> Screenshot 2024-05-07 at 1 38 49 PM I see two border here, can we have only one?

The blue "border" is the outline from keyboard navigation. I've opted for the same style, offset by 2px outside of the element. Technically, there's 2 borders on buttons, selected cards etc. but that's the point of a keyboard nav indicator. This is not an issue, it is expected.

File: libs/blocks/quiz-entry/quiz-entry.css (hunk 13 above)
Comment (colloyd) [reply libs/blocks/quiz-entry/quiz-entry.css:174]: This is the same arrow svg that is used on all milo carousels, it is right there in the reference. It is better for us to standardize this, than create a one-off.

File: libs/blocks/quiz-entry/quiz-entry.css (hunk 14 above)
Comment (colloyd) [reply libs/blocks/quiz-entry/quiz-entry.css:81]: The position has been increased so it matches the search icon position more closely.

**Comment by JackySun9** [comment]
> > Screenshot 2024-05-07 at 1 38 49 PM I see two border here, can we have only one?
> 
> The blue "border" is the outline from keyboard navigation. I've opted for the same style, offset by 2px outside of the element. Technically, there's 2 borders on buttons, selected cards etc. but that's the point of a keyboard nav indicator. This is not an issue, it is expected.

You check this, https://helpx.adobe.com/search.html, I like the clear border

File: libs/blocks/quiz-entry/quiz-entry.css [hunk 15]
```diff
@@ -191,45 +284,59 @@ html[dir="rtl"] .quiz-option-icon {
...
+  width: 100%;
+  z-index: -1;
+}
+
+.quiz-option.selected:focus-visible::after {
+  border-radius: 8px;
+  outline-offset: 8px;
 }
 
 .quiz-option.disabled {
```
Comment (JackySun9) [inline libs/blocks/quiz-entry/quiz-entry.css:342]: for some reason, cursor: not-allowed is not working

**Review by JackySun9 — CHANGES_REQUESTED** [review]

**Comment by JackySun9** [comment]
Screenshot 2024-05-07 at 2 45 38 PM
The quiz-directions is not changed when we switch to tablet view

**Comment by JackySun9** [comment]
Screenshot 2024-05-07 at 2 56 24 PM
under mobile view, I only got two cards, per design, we will show the same as tablet and desktop

**Comment by colloyd** [comment]
> Screenshot 2024-05-07 at 2 45 38 PM The quiz-directions is not changed when we switch to tablet view

To be clear, I did not change anything about how the carousel functions, the number of cards shown at different device widths and the core functionality are unchanged. In short, these are not bugs related to this pr.

File: libs/blocks/quiz-entry/quiz-entry.css (hunk 15 above)
Comment (colloyd) [reply libs/blocks/quiz-entry/quiz-entry.css:342]: This issue is not with the css, but with the disabled state of the cards themselves. This seems to have been introduced with @fullcolorcoder's carousel pr, as you can see it happening even on the before URL. We should probably log a bug for this, or Brad can address it in hit carousel css refinement ticket.

**Comment by colloyd** [comment]
> > > Screenshot 2024-05-07 at 1 38 49 PM I see two border here, can we have only one?
> > 
> > 
> > The blue "border" is the outline from keyboard navigation. I've opted for the same style, offset by 2px outside of the element. Technically, there's 2 borders on buttons, selected cards etc. but that's the point of a keyboard nav indicator. This is not an issue, it is expected.
> 
> You check this, https://helpx.adobe.com/search.html, I like the clear border

Addressed the border-offset on the field so it's only got one border. I gave similar treatment to the carousel arrows.

**Review by JackySun9 — APPROVED** [review]

**Pull request merged**